### PR TITLE
Signature tests for DataStreamReader

### LINKED ISSUE #295: [FEATURE REQUEST]:  DataStreamReader E2E tests
**Is your feature request related to a problem? Please describe.**
`DataStreamReader` does not currently have an E2E test suite like `DataStreamWriter` does.

**Describe the solution you'd like**
A `DataStreamReaderTests` suite that validates the `DataStreamReader` method signatures like we do in other test suites for their respective classes.

### PR DESCRIPTION
**Pull request by AFFogarty**
This PR implements signature tests for `DataStreamReader`.  Fixes #295.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [x] You have included any necessary tests in the same PR.

### CODE REVIEW

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            Assert.IsType<DataStreamReader>(dsr.Option("key", "value"));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", true));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", long.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", double.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Options(new Dictionary<string, string>()));
+
+            DataFrame data = _spark.Range(0, 5);
+            using (var tempDirectory = new TemporaryDirectory())
+            {
+                string path = Path.Combine(tempDirectory.Path, Guid.NewGuid().ToString());
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]:
```suggestion
                string filepath = Path.Combine(tempDirectory.Path, Guid.NewGuid().ToString());
```

Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Changed.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+                data.Write().Mode("overwrite").Orc(filePath);
+                Assert.IsType<DataFrame>(dsr.Orc(filePath));
+
+                data.Write().Mode("overwrite").Parquet(filePath);
+                Assert.IsType<DataFrame>(dsr.Parquet(filePath));
+            }
+
+            // Text is a special case because we can't use Range() data.
+            string textFilePath = Path.GetFullPath("Resources/people.txt");
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]:
```suggestion
            string textFilePath = Path.GetFullPath($"{TestEnvironment.ResourceDirectory}people.txt");
```

Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Changed.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            Assert.IsType<DataStreamReader>(dsr.Option("key", long.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", double.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Options(new Dictionary<string, string>()));
+
+            DataFrame data = _spark.Range(0, 5);
+            using (var tempDirectory = new TemporaryDirectory())
+            {
+                string filePath = Path.Combine(tempDirectory.Path, Guid.NewGuid().ToString());
+
+                data.Write().Json(filePath);
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: can you check `DataFrameReaderTests.cs`? I believe we already have these files checked in such that you don't need to create these.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            DataStreamReader dsr = _spark.ReadStream();
+
+            Assert.IsType<DataStreamReader>(dsr.Format("parquet"));
+            Assert.IsType<DataStreamReader>(dsr.Schema("columnName bigint"));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", "value"));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", true));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", long.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Option("key", double.MaxValue));
+            Assert.IsType<DataStreamReader>(dsr.Options(new Dictionary<string, string>()));
```
[the remark this answers is not in the document]
Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Added

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs (hunk 3 above)
Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Changed to use the reference files.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            Assert.IsType<DataStreamReader>(dsr.Options(new Dictionary<string, string>() {
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: nit.  I like [this formatting](https://github.com/dotnet/spark/blob/a7146cb7ce044d74db50111e63b25ba3a9331833/src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameReaderTests.cs#L38)

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+        /// <summary>
+        /// Test signatures for APIs up to Spark 2.3.*.
+        /// </summary>
+        [Fact]
+        public void TestSignaturesV2_3_X()
+        {
+            DataStreamReader dsr = _spark.ReadStream();
+
+            Assert.IsType<DataStreamReader>(dsr.Format("parquet"));
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]:
```suggestion
            Assert.IsType<DataStreamReader>(dsr.Format("parquet"));

```

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            Assert.IsType<DataStreamReader>(dsr.Schema("columnName bigint"));
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]:
```suggestion
            Assert.IsType<DataStreamReader>(dsr.Schema("columnName bigint"));

```

Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Changed

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs (hunk 6 above)
Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Changed.

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs (hunk 5 above)
Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamReaderTests.cs]: Implemented

**Pull request merged**